app/src/main/java/com/gathering/android/signUp/model/AuthRepository.kt

```diff
@@ -0,0 +1,38 @@
+package com.gathering.android.signUp.model
+
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+import javax.inject.Inject
+
+class AuthRepository @Inject constructor() {
+
+    private lateinit var auth: FirebaseAuth
+
+    fun signUpUser(email: String, pass: String, onResponseReady: (ResponseState) -> Unit) {
```

Comment:
Maybe later we can change this callback to a coroutine, however, it's too much for you. later when you learn that we will migrate to coroutine.

Reply:
I am familiar with coroutine and have applied it to some of my projects but for now, I wanted to make it as simple as possible. for sure we will migrate to coroutine. thank you for  your point